packages/semver/src/schematics/ng-add/utils/dependencies.ts

```diff
@@ -0,0 +1,123 @@
+import { chain, noop, Rule, Tree } from '@angular-devkit/schematics';
+import { addDepsToPackageJson, updateJsonInTree } from '@nrwl/workspace';
+import { constants, mkdirSync, writeFileSync } from 'fs';
+
+import { SchemaOptions } from '../schema';
+
+const PACKAGE_JSON = 'package.json';
+
+export interface PackageJson {
+  scripts: PackageJsonPart<string>;
+  devDependencies: PackageJsonPart<string>;
+  config?: {
+    commitizen?: PackageJsonPart<string>;
+  };
+  commitlint: PackageJsonPart<string[]>;
+}
+
+export interface PackageJsonPart<T> {
+  [key: string]: T;
+}
+
+export function addDependencies(options: SchemaOptions) {
+  return options.enforceConventionalCommits
+    ? chain([
+        _addDevDependencies(options),
+        _addCommitizenConfig(),
+        _addCommitlintConfig(),
+        _addHuskyConfig(),
+        _addHuskyConfigMsg(),

```

Comment:
Now if I try to commit just after the schematic I get: 

```
.husky/commit-msg: 2: .: Can't open .husky/_/husky.sh
```

I would suggest adding a final step that installs packages to trigger the `prepare` script to setup husky:


```ts
import { NodePackageInstallTask } from '@angular-devkit/schematics/tasks';

function _installDependencies(): Rule {
  return (tree: Tree, ctx: SchematicContext) => {
    ctx.addTask(new NodePackageInstallTask());
    return tree;
  };
}

export function addDependencies(options: SchemaOptions) {
  return options.enforceConventionalCommits
    ? chain([
        // ...
        _addHuskyConfig(),
        _addHuskyConfigMsg(),
        _installDependencies(), // 👈
      ])
    : noop();
}
```

Reply:
If we do that the packages will be installed twice. The function `_addDevDependencies()` executes the install. I've tested it in Windows and Debian and works.

Reply:
Good point, the `addDepsToPackageJson` does the install under the hood, maybe running `_addDevDependencies` add the end would fix it.